DXMainClient/DXGUI/Generic/LoadingScreen.cs

```diff
@@ -100,71 +104,7 @@ private void Finish()
-            var tunnelHandler = new TunnelHandler(WindowManager, cncnetManager);
-            var privateMessageHandler = new PrivateMessageHandler(cncnetManager, cncnetUserData);
-            
-            var topBar = new TopBar(WindowManager, cncnetManager, privateMessageHandler);
-
-            var optionsWindow = new OptionsWindow(WindowManager, gameCollection, topBar);
-
-            var pmWindow = new PrivateMessagingWindow(WindowManager,
-                cncnetManager, gameCollection, cncnetUserData, privateMessageHandler);
-            privateMessagingPanel = new PrivateMessagingPanel(WindowManager);
-
-            var cncnetGameLobby = new CnCNetGameLobby(WindowManager,
-                "MultiplayerGameLobby", topBar, cncnetManager, tunnelHandler, gameCollection, cncnetUserData, mapLoader, discordHandler, pmWindow);
-            var cncnetGameLoadingLobby = new CnCNetGameLoadingLobby(WindowManager, 
-                topBar, cncnetManager, tunnelHandler, mapLoader.GameModes, gameCollection, discordHandler);
-            var cncnetLobby = new CnCNetLobby(WindowManager, cncnetManager, 
-                cncnetGameLobby, cncnetGameLoadingLobby, topBar, pmWindow, tunnelHandler,
-                gameCollection, cncnetUserData, optionsWindow);
-            var gipw = new GameInProgressWindow(WindowManager);
-
-            var skirmishLobby = new SkirmishLobby(WindowManager, topBar, mapLoader, discordHandler);
-
-            topBar.SetSecondarySwitch(cncnetLobby);
-
-            var mainMenu = new MainMenu(WindowManager, skirmishLobby, lanLobby,
-                topBar, optionsWindow, cncnetLobby, cncnetManager, discordHandler);
             WindowManager.AddAndInitializeControl(mainMenu);
-
-            DarkeningPanel.AddAndInitializeWithControl(WindowManager, skirmishLobby);
-
-            DarkeningPanel.AddAndInitializeWithControl(WindowManager, cncnetGameLoadingLobby);
-
-            DarkeningPanel.AddAndInitializeWithControl(WindowManager, cncnetGameLobby);
-
-            DarkeningPanel.AddAndInitializeWithControl(WindowManager, cncnetLobby);
-
-            DarkeningPanel.AddAndInitializeWithControl(WindowManager, lanLobby);
-
-            DarkeningPanel.AddAndInitializeWithControl(WindowManager, optionsWindow);
-
-            WindowManager.AddAndInitializeControl(privateMessagingPanel);
-            privateMessagingPanel.AddChild(pmWindow);
-
-            topBar.SetTertiarySwitch(pmWindow);
-            topBar.SetOptionsWindow(optionsWindow);
-
-            WindowManager.AddAndInitializeControl(gipw);
-            skirmishLobby.Disable();
-            cncnetLobby.Disable();
-            cncnetGameLobby.Disable();
-            cncnetGameLoadingLobby.Disable();
-            lanLobby.Disable();
-            pmWindow.Disable();
-            optionsWindow.Disable();
-
-            WindowManager.AddAndInitializeControl(topBar);
-            topBar.AddPrimarySwitchable(mainMenu);
-
```

Comment:
All of the constructor calls above were deleted. They're now done via DI.

All initialization of controls were moved to `MainMenu.cs`. This was necessary due to timing of initialization.